src/tactic/clear.lean

```diff
@@ -0,0 +1,114 @@
+/-
+Copyright (c) 2020 Jannis Limperg. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Jannis Limperg
+-/
+
+/-!
+# A better `clear` tactic
+
+We define `clear'`, an improved version of the standard `clear` tactic. It
+solves the following problem: The `clear` tactic, when called with multiple
+hypotheses, sometimes fails even though all hypotheses could be cleared. This is
+because `clear` is sensitive to the ordering of hypotheses:
+
+```
+example {α : Type} {β : α → Type} (a : α) (b : β a) : unit :=
+  begin
+    clear a b, -- fails
+    exact ()
+  end
+```
+
+When `clear` tries to clear `a`, we still have `b`, which depends on `a`, in the
+context, so the operation fails. We give a tactic `clear'` which recognises this
+and clears `b` before `a`.
+
+## Implementation notes
+
+The tactic is implemented by processing the hypotheses in reverse context order:
+hypotheses occurring later in the context are cleared first. This guarantees
+that the dependency order is observed.
+-/
+
+open native
+
+namespace native.rb_map
+
+  meta def index_map_aux {α} [hlt : has_lt α] [decidable_rel hlt.lt]
```

Comment:
Minor style comment: contrary to old code, nowadays we don't indent code in `namespace`s. So this can go flush left.

Reply:
Thanks for the note. I've changed the indentation in the newest version.